packages/arb-avm-cpp/avm/src/machine.cpp

```diff
@@ -324,6 +322,8 @@ void Machine::runOne() {
         }
     }
 
+    auto startStackSize = m.stack.stacksize();
```

Comment:
This is the size after the instruction has run which is incorrect. It should be the size before the instruction has been run, but after the immediate value (if it exists) has been added to the stack.